orchestrator/api/workspaces.py

```
@@ -233,6 +233,43 @@ async def save_byok_preferences(
+    the platform tool (one whitelist, two doors), merged never replace-blind.
+    """
+    from modules.voice.live_settings import validate_voice_live_update
+
+    workspace = db.query(Workspace).get(ctx.workspace_id)
+    if not workspace:
+        raise HTTPException(status_code=404, detail="Workspace not found")
+
+    try:
+        normalized = validate_voice_live_update(payload.get("voice_live"))
+    except ValueError as exc:
+        raise HTTPException(status_code=400, detail=str(exc))
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)